### PR TITLE
Move release pipeline Prow jobs to use workload identity

### PR DESCRIPTION
**Pull request by chizhg**
I'm not sure if this works or not. Probably we have to use the secret key?

/hold

### CODE REVIEW

File: prow/config/jobs/tools-1.12.yaml [hunk 1]
```diff
@@ -184,19 +184,6 @@ requirement_presets:
       name: cgroup
     - emptyDir: {}
       name: docker-root
-  release:
-    annotations: {}
-    args: null
-    env:
-    - name: COSIGN_KEY
```
Comment (howardjohn) [inline prow/config/jobs/tools-1.12.yaml:191]: is t his replaced somehow?

File: prow/cluster/build/prowjob_service_accounts.yaml [hunk 2]
```diff
@@ -35,3 +35,13 @@ metadata:
...
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: 202987436673-compute@developer.gserviceaccount.com
+  labels:
+    app.kubernetes.io/part-of: prow-build
+  namespace: test-pods
+  name: rel-pipeline-service-account
```
Comment (howardjohn) [inline prow/cluster/build/prowjob_service_accounts.yaml:47]: Can we add a comment when/why it should be used?

**Comment by chizhg** [comment]
I just realized it's a bit tricky to make this change, since the release jobs are not using the default service account. I'll get back to it if I have more bandwidth. Closing it for now.

**Pull request closed without merging**